main.cpp

```diff
@@ -174,7 +174,10 @@ class CDnsSeedOpts {
         filter_whitelist.insert(NODE_NETWORK_LIMITED | NODE_WITNESS | NODE_BLOOM); // x40c
     }
     if (host != NULL && ns == NULL) showHelp = true;
-    if (showHelp) fprintf(stderr, help, argv[0]);
+    if (showHelp) {
+      fprintf(stderr, help, argv[0]);
+      exit(0);
```

Comment:
I think this is incorrect when the help message is triggered due to host == NULL or ns == NULL.